client/asset/dcr/externaltx.go

```diff
@@ -340,7 +340,12 @@ func (dcr *ExchangeWallet) isOutputSpent(ctx context.Context, output *outputSpen
 
 	bestBlock := dcr.cachedBestBlock()
 	if nextScanHeight >= bestBlock.height {
-		if nextScanHeight > bestBlock.height {
+		// When a transaction is initially found in a block, that block is also
+		// scanned for spending transactions (see findTxInBlock) as an
+		// optimization since the block was pulled. As such, it is normal for
+		// the lastScannedHeight for newly located transactions to be the best
+		// block, so there is no need to warn, only return.
+		if nextScanHeight > bestBlock.height+1 {
```

Comment:
The caller could just not call `isOutputSpent` in certain conditions, but it takes a bunch more work and involved needless locking in places. This is simplest, and you get the `mainchainAncestor` checks just for safety if something changed after locating the txn moments ago.